### PR TITLE
Refactor to support read-only root and also move data to a volume

### PR DESCRIPTION
**Pull request by ibuildthecloud**

### CODE REVIEW

File: agent/Dockerfile [hunk 1]
```diff
@@ -14,9 +13,10 @@ RUN pip install cattle docker-py
 RUN curl -s http://stedolan.github.io/jq/download/linux64/jq > /usr/bin/jq; chmod +x /usr/bin/jq
 RUN curl -s https://get.docker.io/builds/Linux/x86_64/docker-1.3.2 > /usr/bin/docker; chmod +x /usr/bin/docker
```
Comment (cloudnautique) [inline agent/Dockerfile]: Can you make this get docker 1.4.1?

File: agent/Dockerfile [hunk 2]
```diff
@@ -1,5 +1,4 @@
-# Install nsenter move to ubuntu:14.04.1
-FROM ibuildthecloud/ubuntu-core-base:14.04-1
+FROM ubuntu:14.04.1
```
Comment (thaJeztah) [inline agent/Dockerfile]: It's probably better to use `ubuntu:14.04` here, unless you really want to pin to `14.04.1`; if `14.04.2` becomes available, `14.04` will automatically be updated to that, whereas `14.04.1` will not.

Comment (ibuildthecloud) [inline agent/Dockerfile]: The idea is reproducibility.  I'm sure there's multiple opinions here, but the intentions was that if you built the image from the tag it should always yield the same result.

File: agent/Dockerfile (hunk 1 above)
Comment (ibuildthecloud) [inline agent/Dockerfile]: I'm going to do that in a different PR.

File: agent/Dockerfile (hunk 2 above)
Comment (thaJeztah) [inline agent/Dockerfile]: Allright then; I also mentioned it because some people were under the impression that `14.04` would be *pinned* to `14.04.0` whereas it is basically `14.04.*`

Doing it for reproducibility is a good reason to pin it
